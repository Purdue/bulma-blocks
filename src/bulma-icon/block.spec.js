jest.setTimeout(30000)
import {
    createNewPost,
    enablePageDialogAccept,
    getEditedPostContent,
    insertBlock
} from '@wordpress/e2e-test-utils';

import {
    blockStartup,
    clickRadio,
    clickElementByText
} from '../test-helpers'

const block = {blockTitle: 'Bulma - Icon', blockName: 'bulma-blocks/icon'}

describe( `ðŸ”¬ ${block.blockTitle} Block`, () => {
    beforeAll( async () => {
        await enablePageDialogAccept();
    } );
    beforeEach( async () => {
        await createNewPost();
    } );

    test( 'ðŸ”Ž Block should be available.', async () => {
        await insertBlock(block.blockTitle)

        // tests that the block is properly inserted and matches the existing snapshot
        expect(await page.$(`[data-type="${block.blockName}"]`)).not.toBeNull()
        expect( await getEditedPostContent() ).toMatchSnapshot()
    } )

    describe( 'ðŸ”¬ Side Panel Settings', () => {
        describe( 'ðŸ”¬ Radio Settings', () => {
            test( 'ðŸ”Ž Icon Container Size Options', async () => {
                await blockStartup(block)

                // select small container
                await clickRadio('Icon Container Size', 'is-small')
                
                let editedContent = await getEditedPostContent()
                
                expect( editedContent ).toMatchSnapshot()
                expect( editedContent.includes('"container":"is-small"')).toBe(true)

                // select medium container
                await clickRadio('Icon Container Size', 'is-medium')
                
                editedContent = await getEditedPostContent()
                
                expect( editedContent ).toMatchSnapshot()
                expect( editedContent.includes('"container":"is-medium"')).toBe(true)

                // select large container
                await clickRadio('Icon Container Size', 'is-large')
                
                editedContent = await getEditedPostContent()
                
                expect( editedContent ).toMatchSnapshot()
                expect( editedContent.includes('"container":"is-large"')).toBe(true)

                // select regular container
                await clickRadio('Icon Container Size', 'sm')
                
                editedContent = await getEditedPostContent()
                
                expect( editedContent ).toMatchSnapshot()
                expect( editedContent.includes('"container":"is-large"')).toBe(false)
                expect( editedContent.includes('"container":"is-medium"')).toBe(false)
                expect( editedContent.includes('"container":"is-small"')).toBe(false)
            })

            test( 'ðŸ”Ž Icon Size Options', async () => {
                await blockStartup(block)

                // select regular
                await clickRadio('Icon Size', 'fa-lg')
                
                let editedContent = await getEditedPostContent()
                
                expect( editedContent ).toMatchSnapshot()
                expect( editedContent.includes('"icon":"fa-lg"')).toBe(true)

                // select medium
                await clickRadio('Icon Size', 'fa-2x')
                
                editedContent = await getEditedPostContent()
                
                expect( editedContent ).toMatchSnapshot()
                expect( editedContent.includes('"icon":"fa-2x"')).toBe(true)

                // select large
                await clickRadio('Icon Size', 'fa-3x')
                
                editedContent = await getEditedPostContent()
                
                expect( editedContent ).toMatchSnapshot()
                expect( editedContent.includes('"icon":"fa-3x"')).toBe(true)

                // select regular
                await clickRadio('Icon Size', 'sm')
                
                editedContent = await getEditedPostContent()
                
                expect( editedContent ).toMatchSnapshot()
                expect( editedContent.includes('"icon":"fa-3x"')).toBe(false)
                expect( editedContent.includes('"icon":"fa-2x"')).toBe(false)
                expect( editedContent.includes('"icon":"fa-lg"')).toBe(false)
            })
        })
        describe( 'ðŸ”¬ Icon Class and Link Settings', () => {
            test( 'ðŸ”Ž Icon Class Textbox', async () => {
                await blockStartup(block)

                const typeString = "Font Awesome Icon Name"
            
                // focus the text box then type into it with the virtual keyboard
                await clickElementByText('label', 'Font Awesome Icon Name')
                await page.keyboard.type(typeString, {delay: 10})

                const editedContent = await getEditedPostContent()

                expect( editedContent).toMatchSnapshot()
                expect( editedContent.includes(`"faClass":"${typeString}"`)).toBe(true)
            })

            test( 'ðŸ”Ž Icon Optional Link', async () => {
                await blockStartup(block)

                const typeString = "https://www.purdue.edu"
            
                // focus the text box then type into it with the virtual keyboard
                await clickElementByText('label', 'Link (Optional)')
                await page.keyboard.type(typeString, {delay: 10})

                const editedContent = await getEditedPostContent()

                expect( editedContent).toMatchSnapshot()
                expect( editedContent.includes(`"link":"${typeString}"`)).toBe(true)
            })
        })
    })

})