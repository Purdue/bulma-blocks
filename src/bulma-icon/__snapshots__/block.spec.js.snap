// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`🔬 Bulma - Icon Block 🔎 Block should be available. 1`] = `
"<!-- wp:bulma-blocks/icon {\\"icon\\":\\"sm\\",\\"container\\":\\"sm\\"} -->
<div class=\\"icon\\"><a><i class=\\"fas fa-undefined\\" aria-hidden=\\"true\\"></i></a></div>
<!-- /wp:bulma-blocks/icon -->"
`;

exports[`🔬 Bulma - Icon Block 🔬 Side Panel Settings 🔬 Radio Settings 🔎 Icon Container Size Options 1`] = `
"<!-- wp:bulma-blocks/icon {\\"icon\\":\\"sm\\",\\"container\\":\\"is-small\\"} -->
<div class=\\"icon is-small\\"><a><i class=\\"fas fa-undefined\\" aria-hidden=\\"true\\"></i></a></div>
<!-- /wp:bulma-blocks/icon -->"
`;

exports[`🔬 Bulma - Icon Block 🔬 Side Panel Settings 🔬 Radio Settings 🔎 Icon Container Size Options 2`] = `
"<!-- wp:bulma-blocks/icon {\\"icon\\":\\"sm\\",\\"container\\":\\"is-medium\\"} -->
<div class=\\"icon is-medium\\"><a><i class=\\"fas fa-undefined\\" aria-hidden=\\"true\\"></i></a></div>
<!-- /wp:bulma-blocks/icon -->"
`;

exports[`🔬 Bulma - Icon Block 🔬 Side Panel Settings 🔬 Radio Settings 🔎 Icon Container Size Options 3`] = `
"<!-- wp:bulma-blocks/icon {\\"icon\\":\\"sm\\",\\"container\\":\\"is-large\\"} -->
<div class=\\"icon is-large\\"><a><i class=\\"fas fa-undefined\\" aria-hidden=\\"true\\"></i></a></div>
<!-- /wp:bulma-blocks/icon -->"
`;

exports[`🔬 Bulma - Icon Block 🔬 Side Panel Settings 🔬 Radio Settings 🔎 Icon Container Size Options 4`] = `
"<!-- wp:bulma-blocks/icon {\\"icon\\":\\"sm\\",\\"container\\":\\"sm\\"} -->
<div class=\\"icon\\"><a><i class=\\"fas fa-undefined\\" aria-hidden=\\"true\\"></i></a></div>
<!-- /wp:bulma-blocks/icon -->"
`;

exports[`🔬 Bulma - Icon Block 🔬 Side Panel Settings 🔬 Radio Settings 🔎 Icon Size Options 1`] = `
"<!-- wp:bulma-blocks/icon {\\"icon\\":\\"fa-lg\\",\\"container\\":\\"sm\\"} -->
<div class=\\"icon\\"><a><i class=\\"fas fa-lg fa-undefined\\" aria-hidden=\\"true\\"></i></a></div>
<!-- /wp:bulma-blocks/icon -->"
`;

exports[`🔬 Bulma - Icon Block 🔬 Side Panel Settings 🔬 Radio Settings 🔎 Icon Size Options 2`] = `
"<!-- wp:bulma-blocks/icon {\\"icon\\":\\"fa-2x\\",\\"container\\":\\"sm\\"} -->
<div class=\\"icon\\"><a><i class=\\"fas fa-2x fa-undefined\\" aria-hidden=\\"true\\"></i></a></div>
<!-- /wp:bulma-blocks/icon -->"
`;

exports[`🔬 Bulma - Icon Block 🔬 Side Panel Settings 🔬 Radio Settings 🔎 Icon Size Options 3`] = `
"<!-- wp:bulma-blocks/icon {\\"icon\\":\\"fa-3x\\",\\"container\\":\\"sm\\"} -->
<div class=\\"icon\\"><a><i class=\\"fas fa-3x fa-undefined\\" aria-hidden=\\"true\\"></i></a></div>
<!-- /wp:bulma-blocks/icon -->"
`;

exports[`🔬 Bulma - Icon Block 🔬 Side Panel Settings 🔬 Radio Settings 🔎 Icon Size Options 4`] = `
"<!-- wp:bulma-blocks/icon {\\"icon\\":\\"sm\\",\\"container\\":\\"sm\\"} -->
<div class=\\"icon\\"><a><i class=\\"fas fa-undefined\\" aria-hidden=\\"true\\"></i></a></div>
<!-- /wp:bulma-blocks/icon -->"
`;
